src/frontend-samples/screen-space-effects-sample/ScreenSpaceEffectsUI.tsx

```diff
@@ -0,0 +1,153 @@
+    update: (newConfig: EffectsConfig, newValue: number) => void) {
+    const updateValue = (values: readonly number[]) => {
+      this.setState((prev) => {
+        const effectsConfig = cloneEffectsConfig(prev.effectsConfig);
+        update(effectsConfig, values[0]);
+        return { effectsConfig };
+      });
+    };
+
+    return (
+      <>
+        <span>{label}</span>
+        <Slider min={min} max={max} step={step} values={[value]} onUpdate={updateValue} disabled={!this.state[enableIf]}></Slider>
+      </>
+    );
+  }
+
+  private readonly _onIModelReady = () => {
+    IModelApp.viewManager.onViewOpen.addOnce((viewport) => {
+      // The lens distortion effect requires the camera to be enabled. Turn it on if it's not already.
+      let lensAngle = this.state.lensAngle;
+      if (!viewport.view.isCameraEnabled())
+        viewport.turnCameraOn(Angle.createDegrees(lensAngle));
+      else
+        lensAngle = viewport.view.camera.getLensAngle().degrees;
+
+      // The grid is distracting and not useful in read-only apps.
+      viewport.viewFlags.grid = false;
+
+      this.setState({
+        viewport,
+        lensAngle,
+      });
+    });
+  }
+
+  private readonly _onChangeLensAngle = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const lensAngle = Number(event.target.value);
+    this.setState({ lensAngle });
+  }
+
+  public componentDidUpdate(_prevProp: UIProps, prevState: UIState) {
+    const viewport = this.state.viewport;
+    if (!viewport)
+      return;
+
+    let changed = false;
+    if (this.state.lensAngle !== prevState.lensAngle) {
+      changed = true;
+      viewport.turnCameraOn(Angle.createDegrees(this.state.lensAngle));
+    }
+
+    // Was an effect toggled on or off?
+    if (this.state.enableSaturation !== prevState.enableSaturation || this.state.enableVignette !== prevState.enableVignette || this.state.enableLensDistortion !== prevState.enableLensDistortion) {
+      changed = true;
+
+      // Screen-space effects are applied in the order in which they are added to the viewport.
+      // Lens distortion shifts pixels, so we want to apply that first, then saturate, and finally vignette.
+      viewport.removeScreenSpaceEffects();
+      if (this.state.enableLensDistortion)
+        viewport.addScreenSpaceEffect("Lens Distortion");
+
+      if (this.state.enableSaturation)
+        viewport.addScreenSpaceEffect("Saturation");
+
+      if (this.state.enableVignette)
+        viewport.addScreenSpaceEffect("Vignette");
+    }
+
+    // Were the settings controlling the effects changed?
+    if (!equalEffectsConfigs(this.state.effectsConfig, prevState.effectsConfig)) {
+      changed = true;
+      updateEffectsConfig(this.state.effectsConfig);
+    }
+
+    if (changed) {
+      viewport.requestRedraw();
+
+      // ###TODO requestRedraw is supposed to do this, but currently doesn't. Remove once that is fixed.
```

Comment:
In what version will this be included?

Reply:
Fix is currently on master so 2.13. Shall I backport to 2.12?

Reply:
That is not necessary. Just wanted to know when we would need to circle back to this TODO.